src/code/z_play.c

```diff
@@ -540,13 +1997,280 @@ void Play_AssignPlayerActorCsIdsFromScene(GameState* thisx, s32 startActorCsId)
+    R_PLAY_FILL_SCREEN_G = green;
+    R_PLAY_FILL_SCREEN_B = blue;
+    R_PLAY_FILL_SCREEN_ALPHA = alpha;
+}
+
+void Play_Init(GameState* thisx) {
+    PlayState* this = (PlayState*)thisx;
+    GraphicsContext* gfxCtx = this->state.gfxCtx;
+    s32 pad;
+    uintptr_t zAlloc;
+    s32 zAllocSize;
+    Player* player;
+    s32 i;
+    s32 spawn;
+    u8 sceneLayer;
+    s32 scene;
+
+    if ((gSaveContext.respawnFlag == -4) || (gSaveContext.respawnFlag == -0x63)) {
+        if (gSaveContext.eventInf[2] & 0x80) {
+            gSaveContext.eventInf[2] &= (u8)~0x80;
+            STOP_GAMESTATE(&this->state);
+            SET_NEXT_GAMESTATE(&this->state, DayTelop_Init, sizeof(DayTelopState));
+            return;
+        }
+
+        gSaveContext.unk_3CA7 = 1;
+        if (gSaveContext.respawnFlag == ~0x62) {
+            gSaveContext.respawnFlag = 2;
+        }
+    } else {
+        gSaveContext.unk_3CA7 = 0;
+    }
+
+    if (gSaveContext.save.entrance == -1) {
+        gSaveContext.save.entrance = 0;
+        STOP_GAMESTATE(&this->state);
+        SET_NEXT_GAMESTATE(&this->state, TitleSetup_Init, sizeof(TitleSetupState));
+        return;
+    }
+
+    if ((gSaveContext.nextCutsceneIndex == 0xFFEF) || (gSaveContext.nextCutsceneIndex == 0xFFF0)) {
+        scene = ((void)0, gSaveContext.save.entrance) >> 9;
+        spawn = (((void)0, gSaveContext.save.entrance) >> 4) & 0x1F;
+
+        if (gSaveContext.save.weekEventReg[33] & 0x80) {
+            if (scene == ENTR_SCENE_MOUNTAIN_VILLAGE_WINTER) {
+                scene = ENTR_SCENE_MOUNTAIN_VILLAGE_SPRING;
+            } else if (scene == ENTR_SCENE_GORON_VILLAGE_WINTER) {
+                scene = ENTR_SCENE_GORON_VILLAGE_SPRING;
+            } else if (scene == ENTR_SCENE_PATH_TO_GORON_VILLAGE_WINTER) {
+                scene = ENTR_SCENE_PATH_TO_GORON_VILLAGE_SPRING;
+            } else if ((scene == ENTR_SCENE_SNOWHEAD) || (scene == ENTR_SCENE_PATH_TO_SNOWHEAD) ||
+                       (scene == ENTR_SCENE_PATH_TO_MOUNTAIN_VILLAGE) || (scene == ENTR_SCENE_GORON_SHRINE) ||
+                       (scene == ENTR_SCENE_GORON_RACETRACK)) {
+                gSaveContext.nextCutsceneIndex = 0xFFF0;
+            }
+        }
+
+        if (gSaveContext.save.weekEventReg[20] & 2) {
+            if (scene == ENTR_SCENE_SOUTHERN_SWAMP_POISONED) {
+                scene = ENTR_SCENE_SOUTHERN_SWAMP_CLEARED;
+            } else if (scene == ENTR_SCENE_WOODFALL) {
+                gSaveContext.nextCutsceneIndex = 0xFFF1;
+            }
+        }
+
+        if ((gSaveContext.save.weekEventReg[52] & 0x20) && (scene == ENTR_SCENE_IKANA_CANYON)) {
+            gSaveContext.nextCutsceneIndex = 0xFFF2;
+        }
+
+        if ((gSaveContext.save.weekEventReg[55] & 0x80) &&
+            ((scene == ENTR_SCENE_GREAT_BAY_COAST) || (scene == ENTR_SCENE_ZORA_CAPE))) {
+            gSaveContext.nextCutsceneIndex = 0xFFF0;
+        }
+
+        if (INV_CONTENT(ITEM_OCARINA) != ITEM_OCARINA) {
```

Comment:
```suggestion
        // Not "First cycle" Termina Field
        if (INV_CONTENT(ITEM_OCARINA) != ITEM_OCARINA) {
```
maybe. Indeed, this entire section is for loading the alternatives based on conditions being satisfied.

Reply:
Hang on, wrong way round: not having Ocarina *is* First cycle...